### DISCUSSION TITLE
[Question] nginx proxy oap server with http2 and ssl

### DISCUSSION THREAD

**Discussion by zzzzl13** [apache/skywalking#10591] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Python Agent (apache/skywalking-python)

### What happened

I use nginx to proxy skywalking oap server and try to connect in python agent. But somehow it comes with this exception, seems the agent consider it as a http1 server. Not sure if I set it wrong. It works fine when I not use ssl in nginx
nginx conf:

```
server {
        listen 8070 http2 ssl;
        ssl_certificate ssl.crt;
        ssl_certificate_key ssl.key;
        server_name localhost;
        access_log  D:/access.log;
        location / {
            grpc_pass grpc://skywalking-oap-server:11800;
            grpc_set_header Host $host;
            grpc_set_header X-Real-IP $remote_addr;
            grpc_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }
        
    }
```
python agent config:
`config.init(agent_collector_backend_services='localhost:8070', agent_protocol='grpc', agent_name='test', agent_instance_name='grpc_test')`

error:
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.UNAVAILABLE
        details = "failed to connect to all addresses; last error: INTERNAL: ipv4:127.0.0.1:8070: Trying to connect an http1.x server"
        debug_error_string = "UNKNOWN:failed to connect to all addresses; last error: INTERNAL: ipv4:127.0.0.1:8070: Trying to connect an http1.x server {created_time:"2023-03-24T03:42:26.522036003+00:00", grpc_status:14}" 

### What you expected to happen

I expect it could connect to the oap server without error

### How to reproduce

use the setting I post above

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Superskyyy** [comment]
As far as I can tell, the NGINX setup looks correct and Python agent merely uses gRPC client to connect to a grpc server without other modifications. 

Please check few things: 

0. ngx_http_v2_module is installed in your nginx deployment, its not a default thing.
1. make sure the ssl certs are valid, or simply remove them and test if things can be connected first.

**Comment by zzzzl13** [comment]
0. ngx_http_v2 module is installed
```
nginx -V
nginx version: nginx/1.22.1
built by cl 16.00.30319.01 for 80x86
built with OpenSSL 1.1.1q  5 Jul 2022
TLS SNI support enabled
configure arguments: --with-cc=cl --builddir=objs.msvc8 --with-debug --prefix= --conf-path=conf/nginx.conf --pid-path=logs/nginx.pid --http-log-path=logs/access.log --error-log-path=logs/error.log --sbin-path=nginx.exe --http-client-body-temp-path=temp/client_body_temp --http-proxy-temp-path=temp/proxy_temp --http-fastcgi-temp-path=temp/fastcgi_temp --http-scgi-temp-path=temp/scgi_temp --http-uwsgi-temp-path=temp/uwsgi_temp --with-cc-opt=-DFD_SETSIZE=1024 --with-pcre=objs.msvc8/lib/pcre2-10.39 --with-zlib=objs.msvc8/lib/zlib-1.2.12 **--with-http_v2_module** --with-http_realip_module --with-http_addition_module --with-http_sub_module --with-http_dav_module --with-http_stub_status_module --with-http_flv_module --with-http_mp4_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_auth_request_module --with-http_random_index_module --with-http_secure_link_module --with-http_slice_module --with-mail --with-stream --with-openssl=objs.msvc8/lib/openssl-1.1.1q --with-openssl-opt='no-asm no-tests -D_WIN32_WINNT=0x0501' --with-http_ssl_module --with-mail_ssl_module --with-stream_ssl_module
```
1. have tested without ssl, just http2, can connect to the skywalking oap server. And for ssl certs, I don't know how to define if it is valid, but have tested in https and it works fine.
